lib/core/src/sync/mod.rs

```diff
@@ -208,6 +227,124 @@ impl SyncService {
 
         Ok(())
     }
+
+    async fn push(&self) -> Result<()> {
+        let outgoing_details = self.persister.get_sync_outgoing_details()?;
+
+        for SyncOutgoingDetails {
+            record_id,
+            record_type,
+            ..
+        } in outgoing_details
+        {
+            // Step 1: Get the sync state, ensure it exists
+            let Some(sync_state) = unwrap_or_continue!(
+                self.persister.get_sync_state_by_record_id(&record_id),
+                "Could not retrieve sync state: {}"
+            ) else {
+                log::debug!("Cannot sync record without any sync state. Skipping.");
+                continue;
+            };
+
+            // Step 2: Fetch the sync data
+            let sync_data = unwrap_or_continue!(
+                self.load_sync_data(&sync_state.data_id, record_type),
+                "Could not load sync data: {}"
+            );
+
+            // Step 3: Create the record to push outwards
+            let record = unwrap_or_continue!(
+                Record::new(
+                    sync_state.record_id.clone(),
+                    sync_data,
+                    sync_state.record_revision,
+                    self.signer.clone()
+                ),
+                "Could not create new record: {}"
+            );
+
+            // Step 5: Push the record
+            let req = unwrap_or_continue!(
+                SetRecordRequest::new(record, utils::now(), self.signer.clone()),
+                "Could not create SetRecordRequest: {}"
+            );
+            let reply =
+                unwrap_or_continue!(self.client.push(req).await, "Could not push record: {}");
+
+            // Step 6: Check for conflict. If present, skip and retry on the next call
+            if reply.status() == SetRecordStatus::Conflict {
+                continue;
+            }
+
+            // Step 7: Update state revision
+            unwrap_or_continue!(
+                self.persister.set_sync_state(SyncState {
+                    data_id: sync_state.data_id,
+                    record_id: sync_state.record_id,
+                    record_revision: reply.new_revision,
+                    is_local: sync_state.is_local,
+                }),
+                "Could not set final sync state after push: {}"
+            );
+        }
+
+        Ok(())
+    }
+
+    pub(crate) async fn enqueue_record(
```

Comment:
This method is racy. The updated_fields should be updated in the same transaction that we update the swap.